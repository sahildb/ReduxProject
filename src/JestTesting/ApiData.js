const axios = require('axios');

// const getData = async () => {
//   const res = await axios.get('https://reqres.in/api/users/2');
//   return res.data.data.id;
// };

const getData = async () => {
  const res = await axios.get('https://reqres.in/api/users/2');
  return res.data;
};
module.exports = {
  getData,
};
// import {StyleSheet, Text, View} from 'react-native';
// import React, {useEffect} from 'react';
// import axios from 'axios';

// const ApiData = () => {
//   const response = axios.get('https://reqres.in/api/users/2');

//   useEffect(() => {
//     response.then(res => {
//       console.log('res', JSON.stringify(res.data.data.id));
//     });
//   });
//   return (
//     <View>
//       <Text>ApiData</Text>
//     </View>
//   );
// };

// export default ApiData;

// const styles = StyleSheet.create({});
