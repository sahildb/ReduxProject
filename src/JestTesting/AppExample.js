import {StyleSheet, Text, TouchableOpacity, View} from 'react-native';
import React from 'react';

const AppExample = () => {
  const btnPress = () => {
    console.log('hello');
  };
  return (
    <View>
      <Text>AppExample</Text>
      <TouchableOpacity onPress={btnPress}>
        <Text> hii</Text>
      </TouchableOpacity>
    </View>
  );
};

export default AppExample;

const styles = StyleSheet.create({});
