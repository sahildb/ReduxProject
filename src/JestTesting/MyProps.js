import {Text, StyleSheet, View} from 'react-native';
import React, {Component} from 'react';

export default class MyProps extends Component {
  constructor(props) {
    super();
  }
  render() {
    return (
      <View>
        <Text>{this.props.name}</Text>
      </View>
    );
  }
}

const styles = StyleSheet.create({});
