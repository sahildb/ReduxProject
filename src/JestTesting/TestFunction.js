const TestFunction = x => {
  return x * 10;
};

export default TestFunction;
