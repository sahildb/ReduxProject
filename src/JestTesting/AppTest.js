//simple function component //
// import {StyleSheet, Text, View} from 'react-native';
// import React from 'react';

// const AppTest = () => {
//   return (
//     <View>
//       <Text>AppTest Add</Text>
//     </View>
//   );
// };

// export default AppTest;

// import React from 'react';
// import {View, Text} from 'react-native';

// export function myFunction(a, b) {
//   return a + b;
// }

// const AppTest = () => {
//   return (
//     <View>
//       <Text>AppTest</Text>
//     </View>
//   );
// };

// export default AppTest;

// simple class function //
// import {Text, View} from 'react-native';
// import React, {Component} from 'react';

// export class AppTest extends Component {
//   render() {
//     return (
//       <View>
//         <Text>AppTest</Text>
//       </View>
//     );
//   }
// }

// export default AppTest;

//function testing
// import {Text, View} from 'react-native';
// import React, {Component} from 'react';

// export class AppTest extends Component {
//   getData(x) {
//     return x + 10;
//   }
//   render() {
//     return (
//       <View>
//         <Text>AppTest</Text>
//       </View>
//     );
//   }
// }

// export default AppTest;
//const styles = StyleSheet.create({});
import {Text, StyleSheet, View} from 'react-native';
import React, {Component} from 'react';
import MyProps from './MyProps';

export default class AppTest extends Component {
  constructor(props) {
    super();
    this.state = {
      counter: 0,
    };
  }

  getData(x) {
    this.setState({counter: this.state.counter + x});
  }

  render() {
    return (
      <View style={{flex: 1, justifyContent: 'center', alignItems: 'center'}}>
        <Text style={{fontSize: 18}}>{this.state.counter}</Text>
        <MyProps name={'sahil'} />
      </View>
    );
  }
}

const styles = StyleSheet.create({});
