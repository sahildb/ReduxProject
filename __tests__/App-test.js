import 'react-native';
import React from 'react';
import AppTest from '../src/JestTesting/AppTest';
import {myFunction} from '../src/JestTesting/AppTest';
import renderer from 'react-test-renderer';
import TestFunction from '../src/JestTesting/TestFunction';
import MyProps from '../src/JestTesting/MyProps';
const axios = require('axios');
const {getData} = require('../src/JestTesting/ApiData');
//snapshot case
test('renders correctly', () => {
  const snapshot = renderer.create(<AppTest />).toJSON();
  console.log('snapshot', snapshot);
  expect(snapshot).toMatchSnapshot();
});

// test('myFunction should add two numbers', () => {
//   //Arrange;
//   const a = 3;
//   const b = 4;

//   // Act
//   const result = myFunction(a, b);

//   // Assert
//   expect(result).toBe(7);
// });
// funcation;
// test('funcation testing', () => {
//   const AppRef = renderer.create(<AppTest />).getInstance();
//   expect(AppRef.getData(2)).toBe(12);
//   const input = AppRef.getData(2);
//   console.log('i', input);
// });

//state
// test('state testing', () => {
//   const AppRef = renderer.create(<AppTest />).getInstance();
//   AppRef.getData(2);
//   expect(AppRef.state.counter).toBe(2);
// });

//props
// test('funcation testing', () => {
//   const AppRef = renderer.create(<MyProps name={'sahil'} />).getInstance();
//   expect(AppRef.props.name).toBe('sahil');
// });

//global function
// describe('function testing directly', () => {
//   test('global function testing', () => {
//     const input = TestFunction(3);
//     const output = 30;
//     expect(input).toBe(output);
//   });
// });

//api test
// it('fetch user', async () => {
//   const apiData = await getData();
//   expect(apiData).toBe(2);
// });

//check full data of api
// it('fetch user', async () => {
//   const apiData = await getData();
//   console.log('apiData', apiData);
//   let data = {
//     data: {
//       avatar: 'https://reqres.in/img/faces/2-image.jpg',
//       email: 'janet.weaver@reqres.in',
//       first_name: 'Janet',
//       id: 2,
//       last_name: 'Weaver',
//     },
//     support: {
//       text: 'To keep ReqRes free, contributions towards server costs are appreciated!',
//       url: 'https://reqres.in/#support-heading',
//     },
//   };
//   expect(apiData).toEqual(data);
// });

//.toBe() is used to evaluate primitives like strings, numbers, or booleans.
//.toEqual() is used to evaluate objects
