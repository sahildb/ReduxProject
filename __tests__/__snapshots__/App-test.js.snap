// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <Text
    style={
      {
        "fontSize": 18,
      }
    }
  >
    0
  </Text>
  <View>
    <Text>
      sahil
    </Text>
  </View>
</View>
`;
